src/AppInstallerCLICore/Commands/SettingsCommand.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "SettingsCommand.h"
+#include "Workflows/WorkflowBase.h"
+#include "Resources.h"
+#include <winget/UserSettings.h>
+
+namespace AppInstaller::CLI
+{
+    using namespace Utility::literals;
+    using namespace AppInstaller::Settings;
+
+    using namespace std::string_view_literals;
+
+    std::vector<Argument> SettingsCommand::GetArguments() const
+    {
+        return {};
+    }
+
+    Resource::LocString SettingsCommand::ShortDescription() const
+    {
+        return { Resource::String::SettingsCommandShortDescription };
+    }
+
+    Resource::LocString SettingsCommand::LongDescription() const
+    {
+        return { Resource::String::SettingsCommandLongDescription };
+    }
+
+    std::string SettingsCommand::HelpLink() const
+    {
+        return "https://aka.ms/winget-settings";
+    }
+
+    void SettingsCommand::ExecuteInternal(Execution::Context& ) const
+    {
+        UserSettingsType userSettingType = UserSettings::Instance().GetType();
+
+        if (userSettingType == UserSettingsType::Default)
+        {
+            // Create settings file if it doesn't exist.
+            UserSettings::Instance().CreateFileIfNeeded();
+        }
+        else if (userSettingType == UserSettingsType::Standard)
+        {
+            // Settings file was loaded correctly, create backup.
+            UserSettings::Instance().CreateBackup();
+        }
+
+        std::string filePathUTF8Str = UserSettings::Instance().SettingsFilePath().u8string();
+
+        // Kudos to the terminal team for this work around.
```

Comment:
>// Kudos to the terminal team for this work around. [](start = 8, length = 51)

Does one have to do something special to invoke the "choose which app to use" dialog?  I expected that would just be automatic.

Reply:
It really depends on the version of Windows. Previous version will fail and that's where notepad will be open. Newer versions will show the dialog.

---
In reply to: [442416924](https://github.com/microsoft/winget-cli/pull/436#discussion_r442416924) [](ancestors = 442416924)